.github/workflows/lint-jsonnet.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: lint-jsonnet
+on: [pull_request]
+  # pull_request:
+  #   paths: "production/**"
+
+jobs:
+  check-mixin:
+    name: Check mixin jsonnet files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22.2'
+      - name: setup jsonnet
+        run: |
+          go install github.com/google/go-jsonnet/cmd/jsonnet@v0.20.0
+          go install github.com/google/go-jsonnet/cmd/jsonnetfmt@v0.20.0
+          go install github.com/google/go-jsonnet/cmd/jsonnet-lint@v0.20.0
+          go install github.com/monitoring-mixins/mixtool/cmd/mixtool@16dc166166d91e93475b86b9355a4faed2400c18
+          go install github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb@v0.5.1
+      - name: run linting
+        run: make BUILD_IN_CONTAINER=false lint-jsonnet
```

Comment:
we're [already linting jsonnet](https://github.com/grafana/loki-release/blob/124c4d996f9625478a79f1884465e29ea082d224/.github/workflows/check.yml#L184) in the [check action](https://github.com/grafana/loki/actions/runs/8666525844/job/23767501298) that runs for every PR, so we should remove this check again here

Reply:
IMO we should be moving as much as possible out of the giant `check` action, I can remove the linting from that action in this PR?